packages/react/ds/src/text-input/text-input.tsx

```diff
@@ -1,33 +1,40 @@
 import React from 'react';
+import { ErrorText, ErrorTextProps } from '../error-text/error-text.js';
+import { HintText, HintTextProps } from '../hint-text/hint-text.js';
+import { Label, LabelProps } from '../label/label.js';
 
 // Extend `React.InputHTMLAttributes<HTMLInputElement>` so that
 // the component can accept all the standard attributes and events that an `<input>` element can handle.
 export type TextInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
   ref?: React.Ref<HTMLInputElement>;
-  hasError?: boolean;
   prefix?: React.ReactNode;
   suffix?: React.ReactNode;
   halfFluid?: boolean;
   fullFluid?: boolean;
   characterWidth?: number;
+  label?: LabelProps;
+  hintText?: HintTextProps;
+  errorText?: ErrorTextProps;
 };
```

Comment:
same here.  the hint and error components contain more props than just the text

Reply:
On it!